.github/workflows/update.yml

```diff
@@ -11,42 +11,88 @@ jobs:
 
     runs-on: ubuntu-latest
     name: Upstream cucumber update
+    permissions:
+      contents: read
+      pull-requests: read
     steps:
       - name: Setup PHP
         uses: shivammathur/setup-php@v2
         with:
-          php-version: 8.1
+          php-version: 8.4
           coverage: none
 
-      - uses: actions/checkout@v4
+      - name: Check whether to update or create PR
+        id: check-existing-pr
+        uses: actions/github-script@v7
+        with:
+          script: |
```

Comment:
I realise this is quite hard to follow in unified diff format. It's much easier to view in split diff.

This needs to run *before* the checkout, because it controls which branch we pull - hence why it is inline in the action.yml rather than a separate script. I envisage this eventually becoming a standalone action.